python/plugins/processing/algs/qgis/StatisticsByCategories.py

```diff
@@ -104,11 +104,15 @@ def processAlgorithm(self, parameters, context, feedback):
             value_field = source.fields().at(value_field_index)
         else:
             value_field = None
-        category_field_indexes = [source.fields().lookupField(n) for n in category_field_names]
+        category_field_indexes = list()
 
         # generate output fields
         fields = QgsFields()
-        for c in category_field_indexes:
+        for field_name in category_field_names:
+            c = source.fields().lookupField(field_name)
+            if c == -1:
+                raise QgsProcessingException(self.tr(f'Field "{field_name}" does not exist.'))
```

Comment:
Just a note -- we can't backport this because it uses f strings and these require a newer version of Python then 3.16 requires. (I'm also unsure if they work ok with translations?)

Reply:
```suggestion
                raise QgsProcessingException(self.tr('Field "{field_name}" does not exist.').format(field_name=field_name))
```

Reply:
Fair point